program/shell/errors_to_json.go

```diff
@@ -0,0 +1,47 @@
+package shell
+
+import (
+	"encoding/json"
+
+	"github.com/elves/elvish/eval"
+	"github.com/elves/elvish/parse"
+)
+
+// ErrorInJSON is the data structure for self-defined error
+type ErrorInJSON struct {
+	FileName string `json:"fileName"`
+	Start    int    `json:"start"`
+	End      int    `json:"end"`
+	Message  string `json:"message"`
+}
+
+// SimplifiedErrorInJSON is the data structure for errors which contain limited info
+type SimplifiedErrorInJSON struct {
```

Comment:
This type does not need to be exported.

It can be named `simpleErrorInJSON`.

The doc comment is too vague; suggested alternative:

```go
// An auxiliary struct for converting errors with only a message to JSON.
```

Reply:
will do